### DISCUSSION TITLE
ui.html

### DISCUSSION THREAD

**Discussion by mahesh11T** [zauberzeug/nicegui#1263] [Q&A]
### Question

if html contains <script> it tells to use ui.add_body_html('login.html with read file method'), this is not executing the JavaScript , how to use without using run_javscript()? and also using query how to access button value from html file and add onclick and access its value, like done with ui.button()?

```py
html_content = '''
<!DOCTYPE html>
<html>
<head>
    <title>HTML Button Example</title>
</head>
<body>

<h1>Button Example</h1>

<button id="my_button">Click Me!</button>

</body>
<script>
" run this script while loading with just open(this.html) as f and ui.html(f)"
</script>
</html>
'''

ui.html(html_content)

def on_button_click(button_value):
    print(f'The button value is: {button_value}')

# Use ui.query_selector() to get the button element by its ID.
button1 = ui.query('#my_button')
print(button1)

button = ui.query('body').classes('#my_button')
print(button)

# Add an onclick event handler to the button.
button.onclick = on_button_click
```

**Comment by falkoschindler** [comment]
What are you trying to achieve?

**Reply by mahesh11T** [reply]
I already have .html and .css files, that I want to integrate with nicegui.

**Comment by falkoschindler** [comment]
Your HTML is pretty broken (script tags should go in the body) and inserting a whole HTML page into a NiceGUI page doesn't make much sense. But in general you can add script tags like this:
```py
ui.add_body_html('''
    <script>
    console.log('Hello from JavaScript!');
    </script>
''')
```
This message _is_ printed to the console when the page loads.

But adding an event handler to a plain HTML button is not that easy.

**Reply by mahesh11T** [reply]
Above html I added just for reference, my actual html file is big, in fastapi there is : 

```py
templates = Jinja2Templates(directory="templates")
@app.get('/inference', response_class=HTMLResponse)
async def show_image(request: Request):
    return templates.TemplateResponse("inference.html", {"request": request})
```

In HTML 
```html
<img id="image" src="{{ url_for('show_image')}}" width="10%" />
```

This will run "show_image" Function , something like this is possible?

**Comment by falkoschindler** [comment]
Please clarify your question.
You mentioned embedding HTML in NiceGUI, adding script tags, executing JavaScript on page load, adding event handlers to HTML objects and now you're bringing up FastAPI and Jinja2.

**Reply by mahesh11T** [reply]
Thanks for your reply, read your response again on event handler with plain html, I will refer the .html and .css files given to me by frontend developer and write it from scratch in nicegui python way itself, as it's not possible to add event handler with plain html.
